staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -490,45 +490,51 @@ func (c *Controller) syncPool(ctx context.Context, poolName string) error {
 	logger.V(5).Info("Existing slices", "obsolete", klog.KObjSlice(obsoleteSlices), "current", klog.KObjSlice(currentSlices))
 
 	if pool, ok := resources.Pools[poolName]; ok {
-		// Match each existing slice against the desired slices.
-		// Two slices match if they contain exactly the same
-		// device IDs, in an arbitrary order. Such a matched
-		// slice gets updated with the desired content if
-		// there is a difference.
+		// Match each existing slice against the desired slices. Two slices
+		// "match" if they contain exactly the same device IDs, in an arbitrary
+		// order. As a special case, slices are also considered "matched" in the
+		// scenario where there's a single existing slice and a single desired
+		// slice. Such a "matched" slice gets updated with the desired content
+		// if there is a difference.
 		//
-		// This supports updating the definition of devices
-		// in a slice. Adding or removing devices is done
-		// by deleting the old slice and creating a new one.
+		// In the case where there is more than one existing or desired slices,
+		// adding or removing devices is done by deleting the old slice and
+		// creating a new one. This is primarily a simplification of the code:
+		// to support adding or removing devices from existing slices, we would
+		// have to identify "most similar" slices (= minimal editing distance).
 		//
-		// This is primarily a simplification of the code:
-		// to support adding or removing devices from
-		// existing slices, we would have to identify "most
-		// similar" slices (= minimal editing distance).
+		// In currentSliceForDesiredSlice we keep track of which desired slice
+		// has a matched slice.
 		//
-		// In currentSliceForDesiredSlice we keep track of
-		// which desired slice has a matched slice.
-		//
-		// At the end of the loop, each current slice is either
-		// a match or obsolete.
+		// At the end of the loop, each current slice is either a match or
+		// obsolete.
```

Comment:
Can you minimize the diff such only your changes are visible? In other words, don't reflow lines that you haven't modified.